client/clientservice/src/request_service.cpp

```diff
@@ -203,7 +211,28 @@ void RequestServiceCallData::sendToConcordClient() {
     config.request.span_context = carrier.str();
     client_->send(config, std::move(msg), callback);
   }
+  metrics_.num_incoming_requests++;
+  updateAggregator();
 }
+
+concordMetrics::Component RequestServiceCallData::metrics_component_{
+    concordMetrics::Component("RequestService", std::make_shared<concordMetrics::Aggregator>())};
+
+RequestServiceCallData::Metrics RequestServiceCallData::metrics_{
+    metrics_component_.RegisterAtomicCounter("num_completed_requests", 0),
+    metrics_component_.RegisterAtomicCounter("num_incoming_requests", 0),
+    metrics_component_.RegisterAtomicCounter("num_failed_requests", 0)};
+
+void RequestServiceCallData::updateAggregator() {
+  const auto metrics_update_duration_microsec{std::chrono::microseconds(1s).count()};
```

Comment:
How often have we updated this metric before? 1sec feels too frequent.

Reply:
What do you mean before? The metric that is used in **Trades/s** comes from DAML so I can't tell.
In replica, for example, the default update interval is 600ms (but it's configurable).

What other frequency do you think would be more reasonable?

Reply:
I see that each component has its own update interval and 1 sec is even too much. Let's set it to 500ms and make it configurable.

Reply:
Actually they all have the same update interval.
I made it configurable (config name: metrics_dump_interval_ms). By default it is set to 1000, other suggestion?

Reply:
500ms?